integration_tests/features/steps/fal_steps.py

```diff
@@ -15,6 +21,69 @@ def run_command_step2(context, command):
     _run_command(command)
 
 
+@given("the project 001_flow_run_with_selectors")
+def set_project_folder(context):
+    context.base_dir = reduce(
+        os.path.join, [os.getcwd(), "projects", "001_flow_run_with_selectors"]
+    )
+    context.temp_dir = tempfile.TemporaryDirectory()
+    os.environ["temp_dir"] = context.temp_dir.name
+
+
+@when("fal flow is invoked with following arguements")
+def invoke_fal_flow(context):
+    _clean_target(context)
+    args = context.text.replace("$baseDir", context.base_dir)
+    cli(["fal", "flow", "run"] + args.split(" "))
+
+
+def _clean_target(context):
+    files = glob.glob(reduce(os.path.join, [context.base_dir, "target", "*"]))
+    for f in files:
+        os.remove(f)
+
+
+@then("the following scripts are ran")
+def check_script_results(context):
+    expected_scripts = context.table.headings
+    for script in expected_scripts:
+        _assertScriptFileExists(context, script)
+
+
+@then("no models are calculated")
+def no_models_are_run(context):
+    run_results_path = reduce(
+        os.path.join, [context.base_dir, "target", "run_results.json"]
+    )
+    if exists(run_results_path):
+        data = json.load(open(run_results_path))
+        assert len(data["results"]) == 0
+    else:
+        assert True
+
+
+@then("no scripts are run")
+def no_scripts_are_run(context):
+    assert len(os.listdir(context.temp_dir.name)) == 0
+
+
+@then("the following models are calculated")
+def check_model_results(context):
+    run_results = open(
+        reduce(os.path.join, [context.temp_dir.name, "target", "run_results.json"])
+    )
+    data = json.load(run_results)
+    calculated_results = map(
+        lambda result: result["unique_id"].split(".")[2], data["results"]
+    )
+    unittest.TestCase().assertCountEqual(calculated_results, context.table.headings)
+
+
+@then("there are no errors")
+def no_errors(context):
+    context.temp_dir.cleanup()
```

Comment:
What does this really do? seems unexpected for "there are no errors"

Reply:
yea moved to:

environment.py
```
def after_all(context):
    os.system("rm -rf mock/temp")
    if context.temp_dir:
        context.temp_dir.cleanup()
```